site/frontend/src/pages/detailed-query/utils.ts

```diff
@@ -244,85 +259,66 @@ export function createTableData(
   // Add totals row first
   const totals = profile.totals;
   const totalsDelta = delta?.totals;
-  result.push({
-    isTotal: true,
-    label: totals.label,
+  result.push(createRowData(true, totals, totalsDelta));
+
+  // Add query data rows
+  profile.query_data.forEach((query, idx) => {
+    const queryDelta = delta?.query_data[idx];
+    result.push(createRowData(false, query, queryDelta));
+  });
+
+  return result;
+}
+
+function createRowData(
+  isTotal: boolean,
+  value: ProfileElement,
+  delta: ProfileElementDelta | undefined
```

Comment:
```suggestion
  delta?: ProfileElementDelta
```

Reply:
Yea, It's not the same thing, but it's a bit obscure. Typescript will yell at you if you don't provide `undefined` as argument. in the `|undefined` case.

Whether you passed `undefined` or ommited the argument is observable by looking at the `arguments` object in the function.